### PR TITLE
CIF-1235 - Create the Core CIF Components library pages - for client-side components

### PR DESCRIPTION
**Pull request by herzog31**
## Description

* Added [storybook](https://storybook.js.org/) to `examples/ui.frontend` project.
* Added new dynamic `storybook` clientlib at `/content/core-components-examples/library/commerce/storybook` that is generated during the build. It contains all the static output files of storybook, so it can be delivered by AEM.
* Added `prettier.io` formatting.
* Added introduction section to storybook to describe the general setup for the React components.
* Added documentation for the following components:
    * `ResetPassword`
    * `CartTrigger`
    * `Cart` (minicart + checkout)
* The components use `MockedProvider` from the Apollo React client to mock the GraphQL endpoint. They don't actually talk to the Mock GraphQL servlet.
* Added new navigation items in component library which links to the before-mentioned components.
* Since the stories require some (not exported) GraphQL queries directly from the `react-components` project, it is required to always build with the `fedDev` profile to use the "snapshot" version of the React components. I therefore set the `fedDev` profile active by default.

You can build the whole project by running `mvn clean install -Pinclude-wcm-components-examples` in the `examples` folder.

## Screenshots (if appropriate):

Screenshot 2020-11-16 at 15 06 40
Screenshot 2020-11-16 at 15 07 24

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

**Review by dplaton — APPROVED** [review]

File: examples/ui.frontend/.storybook/main.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
```
Comment (hbacila) [inline examples/ui.frontend/.storybook/main.js:3]: Copyright 2020 for all new files

**Review by hbacila — APPROVED** [review]

**Review by cjelger — COMMENTED** [review]
In the PR description you write
> the stories require some (not exported) GraphQL queries directly from the react-components project, it is recommended to always build with the fedDev profile to use the "snapshot" version of the React components.

Is this "recommended" or will it work if we don't do that? The code is now automatically committed to the components library azure repo and will be built and deployed by CM so we won't have these profiles being set. Will it work?

File: examples/ui.frontend/.storybook/main.js (hunk 1 above)
Comment (herzog31) [reply examples/ui.frontend/.storybook/main.js]: I updated all new files to 2020 👍

**Comment by herzog31** [comment]
@cjelger I updated the description. The storybook build will fail without the `fedDev` profile on. I changed the pom to have it active by default. So the Cloudmanager build should pass.

**Review by cjelger — APPROVED** [review]

**Pull request merged**